registration/manager_test.go

```diff
@@ -141,7 +143,139 @@ func TestSelfRegistrationManager_defaultHTTPCheck(t *testing.T) {
 	}
 }
 
+func TestSelfRegistrationManager_Start(t *testing.T) {
+	t.Parallel()
+	id := "cts-123"
+	manager := &SelfRegistrationManager{
+		service: &service{
+			name: defaultServiceName,
+			id:   id,
+		},
+		logger: logging.NewNullLogger(),
+		// mock client will be set per test case
+	}
+
+	t.Run("start registers, cancel deregisters", func(t *testing.T) {
+		mockClient := new(mocks.ConsulClientInterface)
+		mockClient.On("RegisterService", mock.Anything,
+			mock.MatchedBy(func(r *consulapi.AgentServiceRegistration) bool {
+				return r.ID == id && r.Name == defaultServiceName
+			})).Return(nil)
+		mockClient.On("DeregisterService", mock.Anything,
+			mock.MatchedBy(func(actual string) bool {
+				return actual == id
+			})).Return(nil)
```

Comment:
```suggestion
		mockClient.On("DeregisterService", mock.Anything, id).Return(nil)
```

This one can potentially be switched to a simpler mock?

Reply:
Yup, will review all of them and make changes where needed!